pyannote/audio/models/models.py

```diff
@@ -526,3 +527,129 @@ def intermediate_dimension(self, layer):
+
+        if sincnet is None:
+            sincnet = dict()
+
+        return SincNet.get_alignment(**sincnet)
+
+    supports_packed = False
+
+    @staticmethod
+    def get_resolution(sincnet : Optional[dict] = None, **kwargs) -> Resolution:
+        """Get sliding window used for feature extraction
+
+        Parameters
+        ----------
+        sincnet : dict, optional
+
+        Returns
+        -------
+        sliding_window : `pyannote.core.SlidingWindow` or {`window`, `frame`}
+        """
+
+        if sincnet is None:
+            sincnet = dict()
+
+        return SincNet.get_resolution(**sincnet)
+
+    def init(self,
+             sincnet : Optional[dict] = None,
+             tdnn : Optional[dict] = None,
+             embedding : Optional[dict] = None):
+        """waveform -> SincNet -> RNN [-> merge] [-> time_pool] -> FC -> output
+
+        Parameters
+        ----------
+        sincnet : `dict`, optional
+            SincNet parameters. Defaults to `pyannote.audio.models.sincnet.SincNet`
+            default parameters.
+        tdnn : `dict`, optional
+            Time-Delay neural network parameters.
+            Defaults to `pyannote.audio.models.tdnn.XVectorNet` default parameters.
+        embedding : `dict`, optional
+            Embedding parameters. Defaults to `Embedding` default parameters. This
+            only has effect when model is used for representation learning.
+        """
+
+        n_features = self.n_features
+
+        if sincnet is None:
+            sincnet = dict()
+        self.sincnet = sincnet
+
+        if n_features != 1:
+            raise ValueError('SincNet only supports mono waveforms. '
+                             f'Here, waveform has {n_features} channels.')
+        self.sincnet_ = SincNet(**sincnet)
+        n_features = self.sincnet_.dimension
+
+        if tdnn is None:
+            tdnn = dict()
+        self.tdnn = tdnn
+        self.tdnn_ = XVectorNet(n_features, **tdnn)
+        n_features = self.tdnn_.dimension
+
+        if self.task.is_representation_learning:
+            if embedding is None:
+                embedding = dict()
+            self.embedding = embedding
+            self.embedding_ = Embedding(n_features, **embedding)
+        else:
+            self.linear_ = nn.Linear(n_features, len(self.classes), bias=True)
+            self.activation_ = self.task.default_activation
+
+    def forward(self, waveforms: torch.Tensor, **kwargs) -> torch.Tensor:
+        """Forward pass
+
+        Parameters
+        ----------
+        waveforms : (batch_size, n_samples, 1) `torch.Tensor`
+            Batch of waveforms
+
+        Returns
+        -------
+        output : `torch.Tensor`
+            Final network output or intermediate network output
+            (only when `return_intermediate` is provided).
+        """
+
+        output = self.sincnet_(waveforms)
+
+        return_intermediate = 'segment6' if self.task.is_representation_learning else None
```

Comment:
My understanding is that the current implementation of `SincTDNN` was only designed for chunk-wise representation learning. 

It would be nice to have it support frame-wise sequence labeling tasks as well (like PyanNet does).

This won't prevent me from merging this PR (once other remarks are taken into account) but this is something that we need to keep in mind.

Reply:
Opened issue #290 to remember this.